File: app/src/main/java/starbright/com/projectegg/dagger/module/RepositoryModule.java
/**
 * Created by Andreas on 5/5/2019.
 */

package starbright.com.projectegg.dagger.module;

import dagger.Module;
import dagger.Provides;
import starbright.com.projectegg.dagger.qualifier.LocalDataStore;
import starbright.com.projectegg.dagger.qualifier.RemoteDataStore;
import starbright.com.projectegg.data.AppDataStore;
import starbright.com.projectegg.data.local.AppLocalDataStore;
import starbright.com.projectegg.data.remote.AppRemoteDataStore;

@Module
public class RepositoryModule {

    @Provides
    @LocalDataStore
    AppDataStore provideLocalDataStore(AppLocalDataStore dataStore) {
        return dataStore;
    }

    @Provides
    @RemoteDataStore
    AppDataStore provideRemoteDataStore(AppRemoteDataStore dataStore) {
        return dataStore;
    }
}

File: app/src/main/java/starbright/com/projectegg/dagger/module/RepositoryModule.kt
/**
 * Created by Andreas on 5/5/2019.
 */

package starbright.com.projectegg.dagger.module

import dagger.Module
import dagger.Provides
import starbright.com.projectegg.dagger.qualifier.LocalData
import starbright.com.projectegg.dagger.qualifier.RemoteData
import starbright.com.projectegg.data.AppDataStore
import starbright.com.projectegg.data.local.AppLocalDataStore
import starbright.com.projectegg.data.remote.AppRemoteDataStore

@Module
class RepositoryModule {

    @Provides
    @LocalData
    internal fun provideLocalDataStore(dataStore: AppLocalDataStore): AppDataStore {
        return dataStore
    }

    @Provides
    @RemoteData
    internal fun provideRemoteDataStore(dataStore: AppRemoteDataStore): AppDataStore {
        return dataStore
    }
}
